### PR TITLE
Datahub template

### PR DESCRIPTION
**Pull request by vdkkia**
There's the ability to show/create templates in this branch. It's supposed to be used in the full version of the **single page** (Sample types will be created based on the templates).
Since there are not many things that users can do with the templates in the Seek base, an on/off switch is considered for **_Templates_** in the server admin panel.

### CODE REVIEW

File: app/controllers/templates_controller.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+          format.html { render action: 'edit', status: :unprocessable_entity }
+          format.json { render json: @template.errors, status: :unprocessable_entity}
+        end
+      end
+    end
+  
+    def destroy
+      respond_to do |format|
+      if @template.can_delete? && @template.destroy
+        format.html { redirect_to @template,location: templates_path, notice: 'Template was successfully deleted.' }
```
Comment (stuzart) [inline app/controllers/templates_controller.rb:64]: hardcoded "Template", but I see later on there is a I18n definition that could be used. Same a couple of lines below

File: app/views/templates/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<h1>Editing Template</h1>
```
Comment (stuzart) [inline app/views/templates/edit.html.erb:1]: hard coded "Template" rather than I18n

File: app/views/templates/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<% tab = @tab || 'manual' %>
+
+<h1>New Template</h1>
+
+<p>
+  Templates can be defined manually through the form below or based on repository templates.
```
Comment (stuzart) [inline app/views/templates/new.html.erb:6]: hard coded "Template" rather than I18n

**Review by stuzart — CHANGES_REQUESTED** [review]
I found a few cases where the t('template') locale wasn't being used, and hard-coded text instead. You might want to check if there any other cases I missed.

There is potential to re-use and combine with the sample-types more, to avoid as much duplication, but this I think it is a later refactoring task once things have settled. 

This is quite a big PR so I will have another scan over later and may add more comments.

File: app/controllers/templates_controller.rb (hunk 1 above)
Comment (vdkkia) [reply app/controllers/templates_controller.rb]: This issue and the other similar issues have been solved via the last commit.

**Pull request merged**